src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/Utilities/DeviceTest.cs

```diff
@@ -0,0 +1,145 @@
+using NUnit.Framework;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Xml;
+using System.Xml.Linq;
+using System.Xml.XPath;
+using Xamarin.ProjectTools;
+using XABuildPaths = Xamarin.Android.Build.Paths;
+
+namespace Xamarin.Android.Build.Tests
+{
+	public class DeviceTest: BaseTest
+	{
+		protected static void RunAdbInput (string command, params object [] args)
+		{
+			RunAdbCommand ($"shell {command} {string.Join (" ", args)}");
+		}
+
+		protected static string ClearAdbLogcat ()
+		{
+			return RunAdbCommand ("logcat -c");
+		}
+
+		protected static string ClearDebugProperty ()
+		{
+			return RunAdbCommand ("shell setprop debug.mono.extra \"\"");
+		}
+
+		protected static void AdbStartActivity (string activity)
+		{
+			RunAdbCommand ($"shell am start -S -n \"{activity}\"");
+		}
+
+		protected void WaitFor (TimeSpan timeSpan, Func<bool> func)
+		{
+			var pause = new ManualResetEvent (false);
+			TimeSpan total = timeSpan;
+			TimeSpan interval = TimeSpan.FromMilliseconds (10);
+			while (total.TotalMilliseconds > 0) {
+				pause.WaitOne (interval);
+				total = total.Subtract (interval);
+				if (func ()) {
+					break;
+				}
+			}
+		}
+
+		protected static bool MonitorAdbLogcat (Func<string, bool> action, int timeout = 10)
+		{
+			string ext = Environment.OSVersion.Platform != PlatformID.Unix ? ".exe" : "";
+			string adb = Path.Combine (AndroidSdkPath, "platform-tools", "adb" + ext);
+			var info = new ProcessStartInfo (adb, "logcat") {
+				RedirectStandardOutput = true,
+				RedirectStandardError = true,
```

Comment:
Why are we redirecting stdout & stderr if we're not using `Process.ErrorDataReceived` or `Process.StandardError`?